### PR TITLE
Add pthread.h to pthread_np.h Autoconf check

### LINKED ISSUE #1306: [openbsd] pthread_np.h needs to be included
This passes at least some first steps in trying to build trafficserver on OpenBSD. Please consider this or better variation, thank you.

```
kern.version=OpenBSD 6.0-current (GENERIC.MP) #0: Wed Nov 23 20:54:10 MST 2016
    build@amd64.openbsd.org:/usr/src/sys/arch/amd64/compile/GENERIC.MP
```

```
--- lib/ts/ink_thread.h.orig    Thu Oct 20 21:08:55 2016
+++ lib/ts/ink_thread.h Sat Jan  7 21:16:56 2017
@@ -44,6 +44,10 @@
 #include <signal.h>
 #include <semaphore.h>

+#if defined(openbsd)
+#include <pthread_np.h>
+#endif
+
 #if HAVE_PTHREAD_NP_H
 #include <pthread_np.h>
 #endif
```

### PR DESCRIPTION
**Pull request by jablko**
@jirib does this fix #1306?

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1321/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1215/ for details.

File: configure.ac [hunk 1]
```diff
@@ -1516,6 +1515,9 @@ AC_CHECK_HEADERS([sys/types.h \
                   editline/readline.h \
                   ucred.h ])
 
+# On OpenBSD, pthread.h must be included before pthread_np.h. #1306
```
Comment (bryancall) [inline configure.ac]: I don't think adding GitHub issues into comments is a good idea.

Comment (jablko) [inline configure.ac]: I don't propose doing it in general, just in this case I don't have another citation for this claim about OpenBSD. If it isn't true for every instance of OpenBSD, someone might question why this is in there, and the issue explains that. There's always "blame", this is just less hassle for that person, I think. There are cases where we reference Jiras from comments, no?

**Comment by bryancall** [comment]
[approve ci]

**Comment by bryancall** [comment]
I created a new PR for this and set the author to Jack Bates  #2283

**Pull request closed without merging**